### ISSUE TITLE
Implement gridsearch from lcmm?

### ISSUE THREAD

**Issue by knokknok** [philips-software/latrend#126]

**Comment by niekdt** [comment]
Thanks for the suggestion. I will implement this in the coming weeks

Note that for repeated fitting with automatic selection of the best result you can already use the [lcFitRepMin](https://philips-software.github.io/latrend/reference/lcFitMethods.html) method modifier. Still, the gridsearch approach will be faster since it does an initial quick exploration for different random starts.

**Comment by niekdt** [comment]
@knokknok I've implemented gridsearch initialization, including parallel computation support. See the `lcmethodLcmmGMM` documentation for details.

Example specification:
```R
  method <- lcMethodLcmmGMM(
    fixed = Y ~ Time,
    mixture = ~ Time,
    random = ~ 1,
    id = "Id",
    time = "Time",
    nClusters = 3,
    init = "gridsearch",
    gridsearch.maxiter = 10,
    gridsearch.rep = 50,
    gridsearch.parallel = TRUE
  )
```

Let me know if you run into any issues.

**Comment by knokknok** [comment]
Thanks!
Out of curiosity, why did you reimplement the grid search functionality instead of calling lcmm's?

**Comment by niekdt** [comment]
Mainly because I already had a custom implementation that I created 4 years ago for a simulation study (see [here](https://github.com/philips-labs/comparison-clustering-longitudinal-data)), which was the basis for creating this package.

I (probably) had a good reason for it at the time but which I don't quite remember haha. Most likely an issue with the `hlme` call evaluation outside of the global environment, or the annoyance of (re)constructing that call dynamically.

Anyways, using the same parallel back-end as the rest of the latrend package is a nice bonus :)

**Comment by niekdt** [comment]
Thanks, I've added this as a test case. Gridsearch init is now ignored for `nClusters = 1` so it's easier to estimate lcmm for 1:N clusters.

**Comment by niekdt** [comment]
Unfortunately I submitted the 1.5.1 CRAN release a couple of hours ago so this fix will not be pushed to CRAN until 2 weeks later at minimum.

**Comment by niekdt** [comment]
Since there was a problem with the CRAN submission this fix will be on the 1.5.1 CRAN release.
